samcli/lib/package/s3_uploader.py

```diff
@@ -263,6 +263,28 @@ def parse_s3_url(
 
         raise ValueError("URL given to the parse method is not a valid S3 url " "{0}".format(url))
 
+    @staticmethod
+    def parse_path_style_s3_url(
```

Comment:
Should we just combine `parse_s3_url` and `parse_path_style_s3_url` into one method to handle both `s3://` and `https://s3` cases? We can still have the two logic separated in two protected methods, but just expose one common method to handle any style of s3 url

Reply:
Yes that makes sense, Mohamed suggested the same thing to me.